server/master/src/main/java/org/apache/accumulo/master/tableOps/bulkVer2/LoadFiles.java

```diff
@@ -328,38 +334,45 @@ private long loadFiles(Table.ID tableId, Path bulkDir, LoadMappingIterator lmi,
     loader.start(bulkDir, master, tid, bulkInfo.setTime);
 
     long t1 = System.currentTimeMillis();
-    while (true) {
-      if (loadMapEntry == null) {
-        if (!lmi.hasNext()) {
-          break;
-        }
-        loadMapEntry = lmi.next();
-      }
-      KeyExtent fke = loadMapEntry.getKey();
-      Files files = loadMapEntry.getValue();
-      loadMapEntry = null;
-
+    while (lmi.hasNext()) {
+      loadMapEntry = lmi.next();
+      KeyExtent fileTablet = loadMapEntry.getKey();
       tablets.clear();
 
-      while (!Objects.equals(currentTablet.getPrevEndRow(), fke.getPrevEndRow())) {
-        currentTablet = tabletIter.next();
-      }
+      // get first tablet in range
+      currentTablet = getFirstTablet(tabletIter, currentTablet, fileTablet.getPrevEndRow());
```

Comment:
I think collapsing these two functions into one would make this code more readable. Then this loop could look like

```java
// note in this code tabletIters is a peeking iter also
    while (lmi.hasNext()) {
      loadMapEntry = lmi.next();
      KeyExtent loadRange = loadMapEntry.getKey();

      List<TabletMetadata> tablets = findOverlapping(loadRange, tabletIter);

      loader.load(tablets, loadMapEntry.getValue());
    }
```

With a find overlapping that looks like the following.

```java
  private List<TabletMetadata> findOverlapping(KeyExtent loadRange,
      PeekingIterator<TabletMetadata> tabletIter) {

    List<TabletMetadata> tablets = new ArrayList<>();

    // skip tablets until have the same prev end row
    while (!Objects.equals(tabletIter.peek().getPrevEndRow(), loadRange.getPrevEndRow())) {
      tabletIter.next();
    }

    // add tablets until the end row matches the load range
    while (true) {
      TabletMetadata tablet = tabletIter.next();
      tablets.add(tablet);
      if (Objects.equals(tablet.getEndRow(), loadRange.getEndRow()))
        break;
    }

    return tablets;
  }
```

Reply:
Actually `tabletIter` does not have to be a peeking iter if the code is written as it currently is.

```java
 private List<TabletMetadata> findOverlapping(KeyExtent loadRange,
      Iterator<TabletMetadata> tabletIter) {

    List<TabletMetadata> tablets = new ArrayList<>();
    TabletMetadata tablet = tabletIter.next();
    
    // skip tablets until have the same prev end row
    while (!Objects.equals(tablet.getPrevEndRow(), loadRange.getPrevEndRow())) {
      tablet = tabletIter.next();
    }
    tablets.add(tablet);
    
    // add tablets until the end row matches the load range
    while(!Objects.equals(tablet.getEndRow(), loadRange.getEndRow())) {
      tablet = tabletIter.next();
      tablets.add(tablet);
    };

    return tablets;
  }
```

Reply:
That is probably a good idea to combine the two into one function with a singular purpose. What about the name "findOverlappingTablets"?  It is a bit longer but states what is being found.

Reply:
`findOverlappingTablets` sounds good.  I was not sure what to name it.